contracts/SpokePool.sol

```diff
@@ -143,6 +143,10 @@ abstract contract SpokePool is
     // token for the input token. By using this magic value, off-chain validators do not have to keep
     // this event in their lookback window when querying for expired deposts.
     uint32 public constant INFINITE_FILL_DEADLINE = type(uint32).max;
+
+    // One year in seconds. If `exclusivityParameter` is set to a value less than this, then the emitted
+    // exclusivityDeadline in a deposit event will be set to the current time plus this value.
+    uint32 public constant MAX_EXCLUSIVITY_PERIOD_SECONDS = 31_536_000;
```

Comment:
I agree with this since no relayer can make a fill on an expired deposit regardless of exclusivity.

On the other hand - having a defined upper limit for whether the `exclusivityParameter` is an offset or timestamp is convenient.

Reply:
I think it's OK to permit a larger relative exclusivityDeadline, I just don't see the use case for a fixed exclusivityDeadline - that's unreliable and the logic to derive whether it's 0, relative or fixed is relatively more complicated as a result.

Reply:
i think there are ways to build an exclusive quoting API with fixed timestamps:
- API gives depositors fixed timestamps to set
- Depositors form deposit and then sign over message and hand off to filler to submit on origin chain
- API rewards fillers who fill exclusively on the destination chain
- API penalizes fillers who do not mine the origin chain transaction on-time

This API also needs to ensure that only well-behaving depositors get to submit orders through this API

IMO this use case is the only realistic way to support exclusivity without exposing fillers to block.timestamp-related re-org risk and without changing the `V3RelayData` struct.

Reply:
I think removing exclusivity deadline from the relay hash is off the table since it means anyone can "steal" exclusivity and it prevents the depositor from offering exclusivity. If we were open to changing the relay hash then I would instead propose adding an initiateDeadline parameter that ensures that deposits don't delay getting mined. 

However changing the relay hash would be highly disruptive so I strongly want to avoid.